pkg/tools/logger/logruslogger.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logger
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func newLogrus(ctx context.Context, entry *logrus.Entry) Logger {
+	logger := &logrusLogger{}
+	ctx = context.WithValue(ctx, ctxKeyLogger, logger)
+	logger.entry = entry.WithContext(ctx)
+	return logger
+}
+
+// NewLogrus - returns a new logrusLogger based on logrus.Entry from context
+func NewLogrus(ctx context.Context) Logger {
```

Comment:
Why we need logrus entry inside context? I supposed we will hold Logger inside context, and we will configure a logger combined with logrus and span processing.